File: sdk/java/src/main/java/com/rmyndharis/openwa/resources/BotConfigResource.java
package com.rmyndharis.openwa.resources;

import static com.rmyndharis.openwa.http.Http.encodeSegment;

import com.rmyndharis.openwa.OpenWAClient;
import com.rmyndharis.openwa.http.HttpMethod;
import java.util.Map;

/** Per-session bot config — access lists, prefix, commands, autoRead, welcome. */
public final class BotConfigResource {
    private final OpenWAClient client;

    public BotConfigResource(OpenWAClient client) {
        this.client = client;
    }

    @SuppressWarnings("unchecked")
    public Map<String, Object> get(String sessionId) {
        return client.request(
            HttpMethod.GET,
            "/api/sessions/" + encodeSegment(sessionId) + "/bot-config",
            null,
            null,
            Map.class);
    }

    @SuppressWarnings("unchecked")
    public Map<String, Object> update(String sessionId, Map<String, Object> body) {
        return client.request(
            HttpMethod.PUT,
            "/api/sessions/" + encodeSegment(sessionId) + "/bot-config",
            null,
            body,
            Map.class);
    }
}

File: sdk/php/src/Resources/BotConfigResource.php
<?php

declare(strict_types=1);

namespace OpenWA\Resources;

use OpenWA\Http\HttpExecutor;

/**
 * Per-session bot config — access lists, prefix, commands, autoRead, welcome.
 *
 * Backed by src/modules/bot/bot-config.controller.ts.
 */
class BotConfigResource
{
    private HttpExecutor $http;

    public function __construct(HttpExecutor $http)
    {
        $this->http = $http;
    }

    /** @return array<string,mixed> */
    public function get(string $sessionId): array
    {
        return $this->http->request('GET', "/api/sessions/{$this->http->encodeSegment($sessionId)}/bot-config");
    }

    /**
     * @param array<string,mixed> $body
     * @return array<string,mixed>
     */
    public function update(string $sessionId, array $body): array
    {
        return $this->http->request(
            'PUT',
            "/api/sessions/{$this->http->encodeSegment($sessionId)}/bot-config",
            [],
            $body
        );
    }
}
